### ISSUE TITLE
Reaching the end of instructions and it says Entities unavailable

### ISSUE THREAD

**Issue by Reachushere** [RogerSelwyn/O365-HomeAssistant#117]
I have followed the instructions and now reviewed them about 50 times and at first I wasn’t able to get the callback link to be accepted in HA at the end, but after figuring it out, I’m now stuck that when it does accept that link, it then says “Entities not available” and the configuration persistent notification disappears.

So I’m not sure what I should be expecting to happen at this point with regards to functionality.  Is an integration supposed to appear that allows me to use features/entities in the Devices section?  I’ve got nothing unfortunately.  I don’t know what entities are unavailable, and it doesn’t say anything about this in the Errors section of the instructions?

Any help would be most appreciated.

**Comment by RogerSelwyn** [comment]
No you don’t get card in the devices section, you only get that for gui setup components. 

Is there any errors showing in the error log?

**Comment by RogerSelwyn** [comment]
Also please post your config, with secrets removed

**Comment by Reachushere** [comment]
I’m not an expert so forgive my nativity, though I’m a relatively quick study… hopefully I’ve done this right, and hopefully this gets posted by replying to you?There are two log errors that say they are connecyef to 0365 and then also this first log error below which I’m not sure what it is so I’m including it just in case it is relevant.My permissions are set as Included in the attached screenshots.  Please don’t hesitate to let me know if I can provide any other information that would be helpful.  Thanks very much for your time!  Have a good day.BrynLogger: aiohttp.serverSource: /usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py:405 First occurred: 7:29:13 AM (1 occurrences) Last logged: 7:29:13 AMError handling requestTraceback (most recent call last):  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 334, in data_received    messages, upgraded, tail = self._request_parser.feed_data(data)  File "aiohttp/_http_parser.pyx", line 551, in aiohttp._http_parser.HttpParser.feed_dataaiohttp.http_exceptions.BadStatusLine: 400, message="Bad status line 'Invalid method encountered'"This error originated from a custom integration.Logger: custom_components.o365Source: custom_components/o365/__init__.py:81 Integration: Office 365 First occurred: January 24, 2023 at 10:13:53 PM (1 occurrences) Last logged: January 24, 2023 at 10:13:53 PMNo token, or token doesn't have all required permissions; requesting authorization for account: Account1This error originated from a custom integration.Logger: custom_components.o365.utilsSource: custom_components/o365/utils.py:174 Integration: Office 365 First occurred: January 24, 2023 at 10:13:53 PM (1 occurrences) Last logged: January 24, 2023 at 10:13:53 PMMinimum required permissions not granted: ['Presence.Read', []]On Jan 25, 2023, at 9:05 AM, RogerSelwyn ***@***.***> wrote:﻿
No you don’t get card in the devices section, you only get that for gui setup components.
Is there any errors showing in the error log?

**Comment by RogerSelwyn** [comment]
Always better to post errors direct in GitHub rather than replying to an email. Then you can Mark them add code using the ‘<>’ in the tool bar. And screenshots aren’t included with an email reply. 

The error tells you you haven’t set all the permissions correctly. You have requested the presence sensor but haven’t set the Presence.Read permission.

**Comment by Reachushere** [comment]
Yes it is a personal account.   Does that mean I can’t use the 0365 integration at all with my personal outlook?On Jan 25, 2023, at 11:24 AM, RogerSelwyn ***@***.***> wrote:﻿
Errors extracted to add some clarity. Please note in particular the last error.
Logger: aiohttp.serverSource: /usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py:405 First occurred: 7:29:13 AM (1 occurrences) Last logged: 7:29:13 AM Error handling requestTraceback (most recent call last):  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 334, in data_received    messages, upgraded, tail = self._request_parser.feed_data(data)  File "aiohttp/_http_parser.pyx", line 551, in aiohttp._http_parser.HttpParser.feed_dataaiohttp.http_exceptions.BadStatusLine: 400, message="Bad status line 'Invalid method encountered'"This error originated from a custom integration.
Logger: custom_components.o365Source: custom_components/o365/__init__.py:81 Integration: Office 365 First occurred: January 24, 2023 at 10:13:53 PM (1 occurrences) Last logged: January 24, 2023 at 10:13:53 PM No token, or token doesn't have all required permissions; requesting authorization for account: Account1 This error originated from a custom integration.
Logger: custom_components.o365.utilsSource: custom_components/o365/utils.py:174 Integration: Office 365 First occurred: January 24, 2023 at 10:13:53 PM (1 occurrences) Last logged: January 24, 2023 at 10:13:53 PM Minimum required permissions not granted: ['Presence.Read', []]

This means you have this added to your config and don't have the correct permission, or it is a personal account:
      status_sensors: # Cannot be used for personal accounts
        - name: "User Teams Status"

**Comment by RogerSelwyn** [comment]
> Yes it is a personal account.

Then you can't use the status and chat sensors..... like it says in the example.

**Comment by Reachushere** [comment]
Ok I really just wanted to integrate excel into some health apps and likely some automations with my emails and calendar, it said that the o365 integration was the only way to do that.  I don’t need a chat.  Sorry, still wrapping my head around a few things here.On Jan 25, 2023, at 12:10 PM, RogerSelwyn ***@***.***> wrote:﻿

Yes it is a personal account.

Then you can't use the status and chat sensors..... like it says in the example.

**Comment by RogerSelwyn** [comment]
Rather than configuring as many things as possible, why not start at the minimum and work from there. Then you know the base works.

**Comment by Reachushere** [comment]
Sure, I just understood that I had to install the office integration as a whole in order to do any of the functions I am looking for.  I’ve got 140 or so smart devices and dozens of great automations I created using the UI.  One area I hadn’t touched yet was office because it seemed to be a different ball game then what I’ve learned so far.On Jan 25, 2023, at 12:38 PM, RogerSelwyn ***@***.***> wrote:﻿
Rather than configuring as many things as possible, why not start at the minimum and work from there. Then you know the base works.
